python/cudnn/gemm/frost/kernel_templates/sm107_moe_grouped_block_scale_matmul_fwd_1ctamma.py

```
@@ -0,0 +1,1365 @@
+                sched_stage = cutlass.Int32(0)
+                sched_full_phase = sched_full_phase ^ 1
+
+            if is_valid != 0:
+                coord_m_group = tile_m * cgrp_tile_mnk[0] + m_rank * cta_tile_mnk[0]
+                coord_n_per_cta = tile_n * cgrp_tile_mnk[1] + n_rank * cta_tile_mnk[1]
+                sfa_m_block = start_sf_block_m + coord_m_group // 128
+                sfb_n_block = coord_n_per_cta // 128
+
+                if group_begin != previous_group_begin:
+                    previous_group_begin = group_begin
+                    for _ai in cutlass.range_constexpr(num_a_operands):
```

Ruff (0.16.1)

[error] Undefined name `num_a_operands`

(F821)